### DISCUSSION TITLE
Migrated from v3 -> v5 and now numeric fields coming out as object in JS for rawQuery

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#21964] [Q&A] [answered]
### Question

I've moved from v3 -> v5 reading both the v4 and v5 migration guides. I'm aware of the change to `queryRaw()` whereby the JS native types are now being used.

That being said, when I run this SQL using query raw (dumbed down version of my main query): 
```sql
select   sum(sr.digitalSaleAmount) as unitsTotal from `sales_report` as `sr` where `sr`.`transactionDate` between '2023-08-01' and '2023-08-31'
```

The type of that column (`sr.digitalSaleAmount`) in the `sales_report` table is of type `int`.

MySQL reports the `unitsTotal` column as `NEWDECIMAL` 
```
Field   1:  `unitsTotal`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       NEWDECIMAL
Collation:  binary (63)
Length:     36
Max_length: 4
Decimals:   0
Flags:      BINARY NUM
```

but when I do `typeof queryRes['unitsTotal']` in JS, it's coming out as an `object` type. 

Log (where key = 'unitsTotal'): `console.log(key, queryItem[key], JSON.stringify(queryItem[key]), typeof queryItem[key])`

Output: 
![image](https://github.com/prisma/prisma/assets/10404747/2b68adc0-5d1f-42bd-94c2-b0ae80bf8333)

How come this is happening? If I log `queryRes['unitsTotal']` I can see the number but because the type is of object, it throws all calculations off I do within JS from that point...

I can cast the field as DOUBLE in my query but I'm hoping to avoid that because I'm using calculations like this a lot and it's a lot of extra work ...

### How to reproduce (optional)

Run a query that returns a DECIMAL and JS returns an object.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model SalesReport {
  id              Int      @id @default(autoincrement())
  transactionDate DateTime
 
  digitalSaleAmount   Int @default(0)
}
```

```ts
select   sum(sr.digitalSaleAmount) as unitsTotal from `sales_report` as `sr` where `sr`.`transactionDate` between '2023-08-01' and '2023-08-31'
```

- OS: Windows
- Database: MYSQL 8
- Node.js version: 16.18.0

- Run `prisma -v` to see your Prisma version and paste it: 5.6.0

**Comment by webnoob** [comment]
I've just thought - the Decimal output isn't a native JS type. I guess this is relating to Prisma's Decimal type i.e Decimal.js but then calling `queryRes['unitsTotal'].abs()` doesn't do anything so I'm not sure.

**Comment by ludralph** [comment, marked as answer]
Hi @webnoob 👋 

Thank you for raising this question.

The result you are observing is the expected behaviour as mentioned in our [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#raw-query-type-mapping). The `Prisma.Decimal` type is not a native Javascript number but it is an Object that represent Decimal numbers with arbitrary precision which is the reason that when you log `typeof queryRes['unitsTotal']` in Javascript, it will show up as an Object because it is an Object. You can use the `toNumber()` method to convert `unitsTotal` to number. 

```js
const unitsTotal = queryRes[0].unitsTotal.toNumber();
console.log('unitsTotal', unitsTotal, typeof unitsTotal)
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hey @webnoob,

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by webnoob** [reply]
Apologies, I got onto something else. Right, this was my deduction so thanks for that. Apologies for not seeing the docs properly first.

**Reply by ludralph** [reply]
No worries 🙌
